File: BookLand_FE/src/types/Notification.ts
import type { User } from './User.ts';

export const NotificationStatus = {
    UNREAD: 'UNREAD',
    READ: 'READ',
    ARCHIVED: 'ARCHIVED'
} as const;

export type NotificationStatus = (typeof NotificationStatus)[keyof typeof NotificationStatus];

export interface Notification {
    id: number;
    from?: User;
    to: User;
    type: string;
    title: string;
    content?: string;
    status: NotificationStatus;
    readAt?: string;
    createdAt?: string;
}

File: BookLand_BE/src/main/java/com/example/bookland_be/entity/Notification.java
package com.example.bookland_be.entity;


import lombok.*;
import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "notification")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Notification {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "fromId")
    private User from;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "toId", nullable = false)
    private User to;

    @Column(nullable = false)
    private String type;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String content;

    @Enumerated(EnumType.STRING)
    @Builder.Default
    private NotificationStatus status = NotificationStatus.UNREAD;

    private LocalDateTime readAt;

    @Column(updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }

    public enum NotificationStatus {
        UNREAD, READ, ARCHIVED
    }
}
